frontend/src/services/session.service.ts

```diff
@@ -1,16 +1,71 @@
 import {Message} from "../models/message";
+import {Artwork} from "../models/artwork";
+import {Game} from "../models/game";
 
 export default class SessionService {
-    public static saveMessages(messages: Message[]): void {
-        localStorage.setItem('messages', JSON.stringify(messages));
+    private static saveInCache<T>(key: string, object: T): void {
+        SessionService.saveTimestamp();
+        localStorage.setItem(key, JSON.stringify(object));
+    }
+
+    private static getFromCache<T>(key: string): T | null {
+        if (SessionService.reloadRequired()) {
+            SessionService.clearCache();
+        }
+        const objectString = localStorage.getItem(key);
+        if (objectString) {
+            return JSON.parse(objectString) as T;
+        }
+        return null;
+    }
+
+    private static saveTimestamp(): void {
+        if (!this.getTimestamp()) {
+            localStorage.setItem('saveDate', Date.now().toString());
+        }
+    }
+
+    private static getTimestamp(): number | null {
+        const savedTimestamp: string | null = localStorage.getItem('saveDate');
```

Comment:
Probably just me since I'm not familiar with typescript, but might be nice to have a comment about how localStorage stores strings. Feel free to ignore if this is common knowledge.